main.go

```diff
@@ -0,0 +1,44 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+
+	"github.com/apache/beam/sdks/v2/go/pkg/beam"
+	"github.com/apache/beam/sdks/v2/go/pkg/beam/x/beamx"
+)
+
+var (
+	input_text = flag.String("input-text", "Default input text", "Input text to print.")
+)
+
+type Result = struct {
```

Comment:
Type definitions don't use an `=`. Using an equal defines a type alias, which isn't what we want here.

```suggestion
type Result struct {
```

Reply:
Secondly, while one *can* define a type for this, one shouldn't. Go would just use multiple returns. 

In this case, I would recommend calling `pipeline, scope := beam.NewPipelineWithRoot()` in `main`, and passing in the scope variable. Then we simply return the PCollection.

Reply:
Got it, that's a great suggestion!